packages/core/factories/src/useUserFactory.ts

```diff
@@ -0,0 +1,113 @@
+import { ref, Ref, watch, computed } from '@vue/composition-api';
+import { UseUser } from '@vue-storefront/interfaces';
+
+export type UseUserFactoryParams<USER, CART, UPDATE_USER_PARAMS> = {
+  cart: Ref<CART>;
+  loadUser: () => Promise<USER>;
+  logOut: (params?: {currentUser?: USER}) => Promise<void>;
+  updateUser: (params: {currentUser: USER; paramsToUpdate: UPDATE_USER_PARAMS}) => Promise<USER>;
+  register: (params: { email: string; password: string; firstName?: string; lastName?: string }) => Promise<USER>;
+  logIn: (params: { username: string; password: string }) => Promise<({updatedUser: USER; updatedCart: CART})>;
```

Comment:
this is useCart job to handle cart
```suggestion
  logIn: (params: { username: string; password: string }) => Promise<({updatedUser: USER})>;
```

Reply:
Done